backend/tests/unit/services/test_mcp_server_registry_masking.py

```
@@ -0,0 +1,388 @@
+        assert hasattr(config, 'data_masking')
+        assert config.data_masking is not None
+        assert config.data_masking.enabled is True
+        
+        # Test non-existent server (should return None, not raise exception)
+        config = registry.get_server_config_safe("non-existent-server")
+        assert config is None
+    
+    def test_masking_config_validation_during_registry_initialization(self):
+        """Test that invalid masking configs fail during registry initialization."""
+        # Invalid masking config should fail during MCPServerConfig creation
+        with pytest.raises(Exception):  # ValidationError from Pydantic
```

Ruff (0.12.2)

Do not assert blind exception: `Exception`

(B017)